staging/src/k8s.io/apiserver/pkg/audit/policy/checker_test.go

```diff
@@ -213,3 +214,144 @@ func TestChecker(t *testing.T) {
+	test("namespaced", audit.LevelRequest, []audit.Stage{audit.StageRequestReceived, audit.StageResponseStarted, audit.StageResponseComplete}, "only audit panic", "getPods", "default")
+
+	test("cluster", audit.LevelMetadata, globalStages, "default")
+	test("cluster", audit.LevelNone, globalStages, "create")
+	test("cluster", audit.LevelMetadata, globalStages, "tims")
+	test("cluster", audit.LevelMetadata, globalStages, "humans")
+	test("cluster", audit.LevelNone, globalStages, "serviceAccounts")
+	test("cluster", audit.LevelNone, globalStages, "getPods")
+	test("cluster", audit.LevelRequestResponse, globalStages, "getClusterRoles")
+	test("cluster", audit.LevelRequest, globalStages, "clusterRoleEdit", "getClusterRoles")
+	test("cluster", audit.LevelNone, globalStages, "getLogs")
+	test("cluster", audit.LevelNone, globalStages, "getMetrics")
+	test("cluster", audit.LevelMetadata, globalStages, "getMetrics", "serviceAccounts", "default")
+	test("cluster", audit.LevelRequestResponse, globalStages, "getMetrics", "getClusterRoles", "default")
+	test("cluster", audit.LevelNone, globalStages, "getPodLogs", "getPods")
+	test("cluster", audit.LevelRequest, globalStages, "omit RequestReceived", "getPods", "default")
+	test("cluster", audit.LevelRequest, []audit.Stage{audit.StageRequestReceived, audit.StageResponseStarted, audit.StageResponseComplete}, "only audit panic", "getPods", "default")
+
+	test("nonResource", audit.LevelMetadata, globalStages, "default")
+	test("nonResource", audit.LevelNone, globalStages, "create")
+	test("nonResource", audit.LevelMetadata, globalStages, "tims")
+	test("nonResource", audit.LevelMetadata, globalStages, "humans")
+	test("nonResource", audit.LevelNone, globalStages, "serviceAccounts")
+	test("nonResource", audit.LevelNone, globalStages, "getPods")
+	test("nonResource", audit.LevelNone, globalStages, "getClusterRoles")
+	test("nonResource", audit.LevelRequestResponse, globalStages, "getLogs")
+	test("nonResource", audit.LevelNone, globalStages, "getMetrics")
+	test("nonResource", audit.LevelMetadata, globalStages, "getMetrics", "serviceAccounts", "default")
+	test("nonResource", audit.LevelRequestResponse, globalStages, "getLogs", "getClusterRoles", "default")
+	test("nonResource", audit.LevelNone, globalStages, "getPodLogs", "getPods")
+	test("nonResource", audit.LevelRequest, globalStages, "omit RequestReceived", "getPods", "default")
+	test("nonResource", audit.LevelRequest, []audit.Stage{audit.StageRequestReceived, audit.StageResponseStarted, audit.StageResponseComplete}, "only audit panic", "getPods", "default")
+
+	test("subresource", audit.LevelRequest, globalStages, "getPodLogs", "getPods")
+	test("subresource", audit.LevelRequest, globalStages, "getPods", "getPodLogs")
+}
+
+// stageEqual returns true if s1 and s2 are super set of each other
+func stageEqual(s1, s2 []audit.Stage) bool {
+	s1 = unionStages(s1, nil)
```

Comment:
If you convert them both to `map[T]bool` and then compare maps, this it not needed

Reply:
Done.